### PR TITLE
Add Cosmos Account Region struct

### PR DESCRIPTION
**Pull request by jay-most**
The AccountRegion struct represents an Azure Cosmos DB database account in a specific region.

- [ X ] The purpose of this PR is explained in this or a referenced issue.
- [ X ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

### CODE REVIEW

File: sdk/cosmos/azcosmos/cosmos_account_region_properties.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+// The AccountRegion struct represents an Azure Cosmos DB database account in a specific region.
+type AccountRegion struct {
+	// Gets the name of the database account in the Azure Cosmos DB service.
+	Name string `json:"Constants.Properties.Name"`
```
Comment (ealsur) [inline sdk/cosmos/azcosmos/cosmos_account_region_properties.go:9]: How does this json deserialization work? `json:....` is meant to map to the JSON payload attribute name.

File: sdk/cosmos/azcosmos/cosmos_account_region_properties.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+// The AccountRegion struct represents an Azure Cosmos DB database account in a specific region.
+type AccountRegion struct {
```
Comment (ealsur) [inline sdk/cosmos/azcosmos/cosmos_account_region_properties.go:7]: Name-wise, we might want to align with the Java Track2 SDK: https://github.com/Azure/azure-sdk-for-java/blob/f8f3980d1e86d3b4983bf38477adc7e07ceab443/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DatabaseAccountLocation.java#L11

**Review by ealsur — COMMENTED** [review]
What is the purpose of this PR and how/where is this type used?

File: sdk/cosmos/azcosmos/cosmos_account_region_properties.go (hunk 1 above)
Comment (ealsur) [reply sdk/cosmos/azcosmos/cosmos_account_region_properties.go:9]: For example, if the Json is `{'name':'xxxxxx'}` then Go's Json comment would be `json: name`. See https://pkg.go.dev/encoding/json#Marshal

File: eng/pipelines/templates/steps/build-test.yml [hunk 3]
```diff
@@ -45,7 +46,7 @@ steps:
       foreach ($td in $testDirs) {
         pushd $td
         Write-Host "##[command]Executing go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic $td | go-junit-report -set-exit-code > report.xml"
-        go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic . | go-junit-report -set-exit-code > report.xml
+        go test -run "^Test" -tags=${{parameters.Tags}} -race -v -coverprofile coverage.txt -covermode atomic . | go-junit-report -set-exit-code > report.xml
```
Comment (seankane-msft) [inline eng/pipelines/templates/steps/build-test.yml:49]: When `Tags` is empty are there any issues with this command? Looks like this is only used for the emulator

**Pull request closed without merging**